Description: Handle EOF during atom parsing as well, in case only coordinates and no vibrations are present. Fix for PR#3530875.
Origin: http://openbabel.svn.sourceforge.net/viewvc/openbabel/openbabel/trunk/src/formats/moldenformat.cpp?r1=4837&r2=4835&pathrev=4837
Author: Michael Banck <mbanck@debian.org>
Last-Update: 2012-05-30
Bug: https://sourceforge.net/tracker/index.php?func=detail&aid=3530875&group_id=40728&atid=428740
Applied-Upstream: yes

--- a/src/formats/moldenformat.cpp	2010-10-22 00:05:08.000000000 +0200
+++ b/src/formats/moldenformat.cpp	2012-05-30 20:07:09.078711354 +0200
@@ -123,9 +123,10 @@
           double factor = 1.; // Angstrom
           if( lineBuffer.find( "AU" ) != string::npos ) factor = BOHR_TO_ANGSTROM; // Bohr
           getline( ifs, lineBuffer );
-          while( lineBuffer.find( "[") == string::npos )
+          while( getline( ifs, lineBuffer ) )
             {
               if( lineBuffer == "" ) continue;
+              if( lineBuffer.find( "[" ) != string::npos ) break;
               istringstream is( lineBuffer );
               string atomName;
               int atomId;
